Check SchnorrCommitments for nil

There is a possible nil dereference in the code that can make the implementation panic. In protocols/cmp/keygen/round3.go, all the incoming parameters are checked against nil, with the exception of body.SchnorrCommitments. If his parameter is nil, the following panic happens:

panic: value method github.com/taurusgroup/multi-party-
↪️sig/pkg/zk/sch.Commitment.Domain called using nil *Commitment
↪️pointer goroutine 12 [running]:
github.com/taurusgroup/multi-party-sig/pkg/zk/sch.(*Commitment) ⌋
↪️
.Domain(0xc000362008?)
<autogenerated>:1 +0x34
github.com/taurusgroup/multi-party-sig/pkg/hash.(*Hash).WriteAny( ⌋
↪️0xc00015dce8, {0xc00015dcf0?, 0x1,
↪️0x0?})
pkg/hash/hash.go:99 +0x41d
github.com/taurusgroup/multi-party-sig/pkg/hash.(*Hash).Decommit( ⌋
↪️0xc0000101c0, {0xc000022280, 0x40, 0xc000119570?},
↪️{0xc0001621e0, 0x20, 0x20}, {0xc00015ddf0, 0x8, 0x8})
pkg/hash/commit.go:109 +0x1b1
github.com/taurusgroup/multi-party-sig/protocols/cmp/keygen.( ⌋
↪️*round3).StoreBroadcastMessage(0xc000119230, {{0xc00013ab29,
↪️0x1}, {0x0, 0x0}, 0x1, {0x7d2160, 0xc0002ceb80}})
protocols/cmp/keygen/round3.go:107 +0x47e
github.com/taurusgroup/multi-party-
↪️
sig/internal/test.Rounds.func2()
internal/test/round.go:106 +0x1bf
golang.org/x/sync/errgroup.(*Group).Go.func1()
pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-
↪️036812b2e83c/errgroup/errgroup.go:57
↪️+0x67
created by golang.org/x/sync/errgroup.(*Group).Go
pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-
↪️036812b2e83c/errgroup/errgroup.go:54
↪️+0x8d
FAIL
↪️github.com/taurusgroup/multi-party-sig/protocols/cmp/keygen
↪️1.694s
FAIL

--- a/protocols/cmp/keygen/round3.go
+++ b/protocols/cmp/keygen/round3.go
@@ -65,7 +65,7 @@ func (r *round3) StoreBroadcastMessage(msg round.Message) error {
 	}
 
 	// check nil
-	if body.N == nil || body.S == nil || body.T == nil || body.VSSPolynomial == nil {
+	if body.N == nil || body.S == nil || body.T == nil || body.VSSPolynomial == nil || body.SchnorrCommitments == nil {
 		return round.ErrNilFields
 	}
 	// check RID length

